src/server/rdb_load.cc

```diff
@@ -1331,15 +1331,15 @@ error_code RdbLoaderBase::ReadObj(int rdbtype, OpaqueObj* dest) {
       iores = ReadStreams(rdbtype);
       break;
     case RDB_TYPE_JSON:
-      iores = ReadJson();
+      iores = ReadGeneric(rdbtype);
```

Comment:
Why don't we move this on line 1313 and let it fallthrough just like `RDB_TYPE_STRING` (Which you did the same :smile: )